docs/insiders/upgrade.md

```diff
@@ -0,0 +1,69 @@
+# How to upgrade
+
+When upgrading Insiders, you should always check the version of Material for
+MkDocs which makes up the first part of the version qualifier, e.g.Insiders
+`4.x.x` is currently based on `8.x.x`:
+
+```
+8.x.x-insiders-4.x.x
+```
+
+If the major version increased, it's a good idea to consult the [upgrade
+guide] and go through the steps to ensure your configuration is up to date and
+all necessary changes have been made. 
+
+  [upgrade guide]: ../upgrade.md
+  [list of tags]: https://github.com/squidfunk/mkdocs-material-insiders/tags
+
+Depending on how you installed and what you want to upgrade to you
+need to run different commands:
+
+=== "pip upgrade to release"
+
+    If you installed Insiders via `pip` and you want to upgrade to a
+    specific release, pick the tag from the [list of tags] and replace 
+    the tag at the end of the URL of the command given below:
+
+    ```
+    pip install --upgrade git+https://${GH_TOKEN}@github.com/squidfunk/mkdocs-material-insiders.git@9.4.2-insiders-4.42.0
+    ```
+
+=== "pip upgrade to latest development"
+
+    If you installed Insiders via `pip` and want to upgrade to the
+    latest development version, run:
+
+    ```
+    pip install --upgrade git+https://${GH_TOKEN}@github.com/squidfunk/mkdocs-material-insiders.git
+    ```
+
+=== "git upgrade"
+
+    If you installed Insiders via `git`, you will first need to check
+    out the version you want to install into your workspace. After
+    this is done, you can run `pip` to install that version.
+
+    You can look up the tags using `git tag --sort -refname` or you
+    can consult the [list of tags]. Then, checkout the tag you want to
+    use by replacing the one given in the command below (twice)and running 
+    it from your workspace:
+
+    ``` 
+    cd mkdocs-material 
+    git checkout tags/9.4.2-insiders-4.42.0 -b 9.4.2-insiders-4.42.0
```

Comment:
I think this can be simplified to:

```
git checkout 9.4.2-insiders-4.42.0
```

That's at least why I use when switching tags.

Reply:
creating the new branch prevents the warning about the 'detached head'. It might also help a user to keep track of what releases they have looked at. The reason I included the `tags/` part is that this is how it is usually described online (yes, I had to look up how to check out a tagged version). It might be a toss-up whether starting with the simplest solution is better or starting with the more complicated one and letting people figure out they can simplify it? Happy to take your guidance here.

Reply:
Thanks for the clarification! Yes, detached head state always sounds kind of scary, especially to git newcomers, but the question is what we're optimizing here for. IMHO, users will rarely need to push new commits into Insiders (which they will not be able to in detached head state), because I think they usually just intend to pin their Insiders version rather than basing more commits off that specific tag. Specifying the tag twice leaves more room for errors, as you need to adjust both tags, so I'm rather leaning towards keeping it simple. I'd say we go for simple now and adjust the instructions once we encounter users with problems.

However, quick googling lead me to:

```
git:master $ git switch -c 9.2.0                     
Switched to a new branch '9.2.0'
git:9.2.0 $ 
```

If you can confirm that this has the same outcome as you suggested, we can just use `switch` ☺️

Reply:
For clarification: `switch` will create a new branch with the same name as the tag, based of the tag.

Reply:
I think that the `-c` option does need the branch name and a starting point, so `tags/<tagname>` would still be needed. The `switch` command does seem preferable to `checkout`, though. Perhaps the instructions should avoid calling the branch the same as the tag. Another option is to use the `--detach` option on `checkout`, which avoids the warning.

Reply:
Ah nice, then let's use `--detach`, if we can omit the repeated tag use.